istioctl/cmd/revision.go

```diff
@@ -0,0 +1,326 @@
+// Copyright Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io"
+	"istio.io/api/operator/v1alpha1"
+	"istio.io/istio/operator/cmd/mesh"
+	operator_istio "istio.io/istio/operator/pkg/apis/istio"
+	iopv1alpha1 "istio.io/istio/operator/pkg/apis/istio/v1alpha1"
+	"istio.io/istio/operator/pkg/manifest"
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/operator/pkg/util/clog"
+	"istio.io/istio/pkg/config"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachinery_schema "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/pointer"
+	"sort"
+	"strings"
+	"text/tabwriter"
+)
+
+type revisionArgs struct {
+	manifestsPath string
+	name          string
+	verbose       bool
+}
+
+var (
+	istioOperatorGVR = apimachinery_schema.GroupVersionResource{
+		Group:    iopv1alpha1.SchemeGroupVersion.Group,
+		Version:  iopv1alpha1.SchemeGroupVersion.Version,
+		Resource: "istiooperators",
+	}
+)
+
+func revisionCommand() *cobra.Command {
+	revisionCmd := &cobra.Command{
+		Use: "revision",
+		Short: "Revision centric view of Istio deployment",
+		Aliases: []string{"rev"},
+	}
+	revisionCmd.AddCommand(revisionListCommand())
+	revisionCmd.AddCommand(revisionDescribeCommand())
+	return revisionCmd
+}
+
+func revisionDescribeCommand() *cobra.Command {
+	describeCmd := &cobra.Command{
+		Use: "describe",
+		Short: "Show details of a revision - customizations, number of pods pointing to it, istiod, gateways etc",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.OutOrStderr().Write([]byte("called revision describe\n"))
+			return nil
+		},
+	}
+	describeCmd.Flags().Bool("all", true, "show all related to a revision")
+	return describeCmd
+}
+
+func revisionListCommand() *cobra.Command {
+	kubeConfigFlags := &genericclioptions.ConfigFlags{
+		Context: pointer.StringPtr(""),
+		Namespace: pointer.StringPtr(""),
+		KubeConfig: pointer.StringPtr(""),
+	}
+	revArgs := revisionArgs{}
+	listCmd := &cobra.Command{
+		Use: "list",
+		Short: "Show list of control plane and gateway revisions that are currently installed in cluster",
```

Comment:
Good to add `Example` to describe how to use this command which will be helpful for doc generation.

Reply:
Added it for each of the subcommands - `list` and `describe`.